### DISCUSSION TITLE
504 Gateway Time-out The server didn't respond in time.

### DISCUSSION THREAD

**Discussion by naman7kr** [grafana/grafana-json-datasource#68] [Q&A] [answered]
I am getting this error when I try to execute a query which takes more than 30s to complete. My json api takes about a minute to execute. Is there a way to set timeout?

Query Inspector ->
request:Object
url:"api/datasources/proxy/1/v1/prog-perf/category-update-dates?program-year=2021"
method:"GET"
hideFromInspector:false
response:Object
error:"Gateway Timeout"
response:"<html><body><h1>504 Gateway Time-out</h1>
The server didn't respond in time.
</body></html>
"
message:"<html><body><h1>504 Gateway Time-out</h1>
The server didn't respond in time.
</body></html>
"

**Comment by naman7kr** [comment]
are you able to reproduce the problem?

**Comment by marcusolsson** [comment]
I haven't had a chance to look at this yet. I should be able to test this during next week. Thanks!

**Comment by marcusolsson** [comment, marked as answer]
Data sources use a proxy provided by Grafana to make requests and typically cannot change the timeout themselves. You should be able to configure it in your grafana.ini:

https://grafana.com/docs/grafana/latest/administration/configuration/#timeout
